### PR TITLE
fix: beacon depends on validator and bridge

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-zkevm-ibc-demo/issues/24

The whole stack now reliably starts

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -11,8 +11,12 @@ services:
...
     entrypoint: /beacon-kit/testing/files/entrypoint.sh
     depends_on:
-      - celestia-network-bridge
-      - reth
+      celestia-network-validator:
+        condition: service_healthy
+      celestia-network-bridge:
+        condition: service_started
+      reth:
+        condition: service_started
```
Comment (cmwaters) [inline docker-compose.yml:19]: I don't think this makes any logical difference as the `celestia-network-bridge` doesn't start until the validator is "healthy" anyway. 

If you wanted to add a health check to the bridge node, you could try using some lightweight method like `/network_head`.

Comment (rootulp) [reply docker-compose.yml:19]: Good point. I tried to add a health check using node.Info or p2p.Info but both required an admin token. Looks like `/network_head` only needs a read auth token so seems a tad easier.

Ref: https://node-rpc-docs.celestia.org/?version=v0.20.4 

Created https://github.com/celestiaorg/celestia-node/issues/4007 so celestia-node exposes an un authenticated status endpoint. In the meantime DevOps shared a workaround that they use:

```shell
exec [sh -c AUTH_TOKEN=$(celestia bridge auth read --node.store=/home/celestia 2>/dev/null); curl --max-time 0.5 -s -X POST -H "Content-type: application/json" -H "Accept: application/json" -H "Authorization: Bearer $AUTH_TOKEN" -d '{"id":1,"jsonrpc":"2.0","method":"header.SyncWait","params":[]}' "http://localhost:26658"; if [ "$?" -eq 0 ]; then exit 0; else echo "Catching up"; exit 1; fi ]
```

so I'll try something like that

**Comment by rootulp** [comment]
A couple of findings:
- We need to use two `$` to reference an environment variable in the healthcheck command. Example: `$$FOO`
- I can't hit the RPC server of the celestia-node instance

**Comment by rootulp** [comment]
Update the celestia bridge node is starting prematurely. If I retry starting it after the validator has been up for some time, it works. However, on initial launch it doesn't work and fails to start the RPC server.

The healthcheck on the validator doesn't seem to be granular enough.

**Review by ninabarbakadze — COMMENTED** [review]
can you please push your branch to the repo because i can't fetch it unless i set your fork as upstream 🙏🏻

**Comment by rootulp** [comment]
You can fetch with this thing without setting my fork as upstream

Screenshot 2024-12-17 at 9 07 35 AM

**Review by cmwaters — APPROVED** [review]
Ran this a couple times and worked for me every time

**Comment by ninabarbakadze** [comment]
> You can fetch with this thing without setting my fork as upstream.
> 
> Update: also pushed this branch to upstream
> 
> Screenshot 2024-12-17 at 9 07 35 AM

thanks! didn't know you could do this

**Review by ninabarbakadze — APPROVED** [review]

**Pull request merged**